crates/router/src/connector/square/transformers.rs

```diff
@@ -298,7 +294,7 @@ impl TryFrom<&types::PaymentsAuthorizeRouterData> for SquarePaymentsRequest {
             | api::PaymentMethodData::GiftCard(_)
             | api::PaymentMethodData::PayLater(_)
             | api::PaymentMethodData::Wallet(_) => Err(errors::ConnectorError::NotImplemented(
-                "Payment Method".to_string(),
+                utils::get_unimplemented_payment_method_error_message("Square"),
```

Comment:
We can club all the `match arms` and propagate the expected common error.
Please use `|` and throw the `NotImplemented` error in the end. No need for it to be done twice

Reply:
Done